webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala

```diff
@@ -1822,24 +1823,32 @@ class StringFormatter private (
    *                  valid for a Knora XML import namespace.
    * @return the corresponding project-specific internal ontology IRI.
    */
+  @deprecated("Use xmlImportNamespaceToInternalOntologyIriV1(String) instead.")
   def xmlImportNamespaceToInternalOntologyIriV1(namespace: String, errorFun: => Nothing): SmartIri =
+    xmlImportNamespaceToInternalOntologyIriV1(namespace).getOrElse(errorFun)
+
+  def xmlImportNamespaceToInternalOntologyIriV1(namespace: String): Option[SmartIri] =
```

Comment:
I think I would prefer it to use Validation rather than Option in such cases: here `None` represents an "illegal" state, but in many places we have `None` represent a "legal" state

Reply:
`None` is neither "legal" nor "illegal", it simply has a value or not. Used as a return type of a function it simply states "the function is able to return a value for some of its input values". An `Option` does not allow to include any reason of why this is.

Before we had a partial function which would throw and exception provided by the caller. The closest return type is an `Option`. That is because before - for values the function was not defined on - the function did not tell anything about the reasons of why it was not defined, i.e. it did not anything to the externally provided excepetion.